src/main/java/racingcar/controller/RacingController.java

```diff
@@ -0,0 +1,55 @@
+package racingcar.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import racingcar.domain.Car;
+import racingcar.service.RacingService;
+import racingcar.util.Converter;
+import racingcar.util.RandomUtilImpl;
+import racingcar.view.InputView;
+import racingcar.view.OutputView;
+
+public class RacingController {
+
+	private final RacingService racingService = new RacingService();
+	private final RandomUtilImpl randomUtil = new RandomUtilImpl();
```

Comment:
```suggestion
	private final RandomUtil randomUtil = new RandomUtilImpl();
```

요건 이렇게 작성할 수도 있죠!

지금 더즈의 구현과 어떤 차이점이 있을까요?

Reply:
질문을 듣고 찾아보던 도중 DI와 인터페이스로 추상화에 대해 조금 알아봤습니다. 타입을 인터페이스로 추상화함으로써 랜덤 숫자 생성 전략이 바뀌더라도 코드 변경이 줄어들고 다양한 의존관계를 맺을 수 있게 된 것 같습니다. 나중에 RandomUtil을 외부에서 주입받게 되었을 땐 런타임 시점에 동적으로 구현체를 갈아끼우는 것도 가능해질 것 같습니다!

Reply:
더즈가 해주신 말씀이 맞아요. 인터페이스를 사용하여 행위를 구현하도록 처리할 경우에는 말씀해주신 이점을 얻을 수 있죠!

지금 구현에 대해서 이야기해보자면, `RandomUtilImpl` 이라는 구현체의 타입으로 객체를 할당하기보다는, `RandomUtil`이라는 인터페이스로 할당하여 `randomUtil`이라는 변수가 인터페이스로써 사용될 수 있도록 하면 좋아요. 추상화 된 인터페이스 타입을 통해 변수를 더 유연하게 사용할 수 있기 때문이에요.